lib/matplotlib/_mathtext.py

```diff
@@ -1690,65 +1690,59 @@ def __init__(self, c, width, state, always=False, char_class=Char):
         self.width = char.width
 
 
-class Ship:
+def ship(ox, oy, box):
     """
     Ship boxes to output once they have been set up, this sends them to output.
 
     Since boxes can be inside of boxes inside of boxes, the main work of `Ship`
     is done by two mutually recursive routines, `hlist_out` and `vlist_out`,
     which traverse the `Hlist` nodes and `Vlist` nodes inside of horizontal
     and vertical boxes.  The global variables used in TeX to store state as it
-    processes have become member variables here.
+    processes have become local variables here.
     """
 
-    def __call__(self, ox, oy, box):
-        self.max_push    = 0  # Deepest nesting of push commands so far
-        self.cur_s       = 0
-        self.cur_v       = 0.
-        self.cur_h       = 0.
-        self.off_h       = ox
-        self.off_v       = oy + box.height
-        self.hlist_out(box)
+    cur_v       = 0.
+    cur_h       = 0.
+    off_h       = ox
+    off_v       = oy + box.height
```

Comment:
Remove spaces if you touch these anyway.
```suggestion
    cur_v = 0.
    cur_h = 0.
    off_h = ox
    off_v = oy + box.height
```

Also in `hlist_out`, `vlist_out`.

Reply:
done